developers-api/src/main/java/com/quarksy/api/DevelopersResource.java

```diff
@@ -30,30 +30,46 @@
     public Response create(Developer developer)
     {
         logger.info("POST Developer");
-        return service.addDev(developer);
+        try {
+            return service.addDev(developer);
+        } catch (Exception e) {
+            throw new WebApplicationException(Response.status(BAD_REQUEST).entity("An error has occurred.").build());
+        }
     }
 
     @GET
     public List<Developer> getAllDevs()
     {
         logger.info("Get All Developers");
-        return service.getAllDevs();
+        try {
+            return service.getAllDevs();
+        } catch (Exception e) {
+            throw new WebApplicationException(Response.status(BAD_REQUEST).entity("An error has occurred.").build());
```

Comment:
I think for this one we should return a 500 Internal Server Error.  Unlikely to be a Bad Request.